config/confighttp/confighttp.go

```diff
@@ -236,9 +236,16 @@ type headerRoundTripper struct {
 
 // RoundTrip is a custom RoundTripper that adds headers to the request.
 func (interceptor *headerRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	// Set Host header if provided
+	hostHeader, found := interceptor.headers["Host"]
+	if found && hostHeader != "" {
+		// `Host` field should be set to override default `Host` header value which is Endpoint
+		req.Host = string(hostHeader)
+	}
 	for k, v := range interceptor.headers {
 		req.Header.Set(k, string(v))
```

Comment:
would Host still be available at this point?

Reply:
hi, yes! It works in test that I wrote and test is broken without this change. I also just tested in my environment and it works.

Reply:
I took a closer look at Go's documentation, and there's one mismatch here compared to that. Go's stdlib precedence is:

- req.Host
- req.URL.Host (from HTTPServerConfig.Endpoint.Host)
- req.Headers["Host"] (from HTTPServerConfig.Headers["Host"])

Right now, we don't have an equivalent to req.Host in confighttp, so this PR introduces the following precedence order:

- HTTPServerConfig.Headers["Host"]
- HTTPServerConfig.Endpoint.Host

While I don't expect this to be a breaking change affecting any users at all given that this header must be explicitly set as part of the headers field, I would prefer if we'd introduce a `HostHeader` field, to be consistent with Go ("path of least surprise" type of thing).

But I'm not 100% convinced myself. I'd appreciate other approvers to give their opinions.

cc @open-telemetry/collector-approvers

Reply:
The `http.Request` doesn't use the Headers["Host"].  

> // For client requests, Host optionally overrides the Host
	// header to send. If empty, the Request.Write method uses
	// the value of URL.Host. Host may contain an international
	// domain name.
	Host [string](https://pkg.go.dev/builtin#string)